charm/ksql/lib/charms/layer/ksql.py

```diff
@@ -60,12 +60,17 @@ def install(self, kafka_units=[]):
         kafka.sort()
         kafka_connect = ','.join(kafka)
 
+        if not state_dir:
+            state_dir = '/var/snap/{}/common/kafka-streams'.format(
```

Comment:
Does this directory need to be created if it does not already exist?

Reply:
@cmars Nope, KSQL seems to already create it if it doesn't exist.